signer/core/apitypes/types.go

```diff
@@ -419,6 +419,39 @@ func (typedData *TypedData) EncodeData(primaryType string, data map[string]inter
 // Attempt to parse bytes in different formats: byte array, hex string, hexutil.Bytes.
 func parseBytes(encType interface{}) ([]byte, bool) {
 	switch v := encType.(type) {
+	case [1]byte:
+	case [2]byte:
+	case [3]byte:
+	case [4]byte:
+	case [5]byte:
+	case [6]byte:
+	case [7]byte:
+	case [8]byte:
+	case [9]byte:
+	case [10]byte:
+	case [11]byte:
+	case [12]byte:
+	case [13]byte:
+	case [14]byte:
+	case [15]byte:
+	case [16]byte:
+	case [17]byte:
+	case [18]byte:
+	case [19]byte:
+	case [20]byte:
+	case [21]byte:
+	case [22]byte:
+	case [23]byte:
+	case [24]byte:
+	case [25]byte:
+	case [26]byte:
+	case [27]byte:
+	case [28]byte:
+	case [29]byte:
+	case [30]byte:
+	case [31]byte:
+	case [32]byte:
+		return v[:], true
```

Comment:
Writing it like this does not cause fall-through. The `[31]byte`-case is an empty clause, so it does nothing, then exits the switch, and returns `nil, false` -- that was not what you intended, was it? 
The way to do a 'fall-through' is 
```
case [32]byte, [31]byte, ... 
```
However, this is a bit clunky. A more elegant solution could probably be made  using `reflect`. 
All in all, I think you would have / should have caught this with a testcase. We do like to make changes to these pretty complex internals unless it's properly backed by testcase.

Reply:
Implemented a different solution using reflect. Added tests for 2, 8 and 16 sized byte arrays, which I think should be enough.